### ISSUE TITLE
Improve and add paths

### ISSUE THREAD

**Issue by banneux-florent** [vantezzen/infinite-craft-solver#3]
Hello, quick question but, how could we make this really collaborative? How could we improve your database and fill it with new content?

As I was curious, I searched for "Submarine":

![image](https://github.com/vantezzen/infinite-craft-solver/assets/10631777/12f41cf9-2005-40c2-b4ee-c1b5e05beefa)

But found a quicker way (I edited the page):

![image](https://github.com/vantezzen/infinite-craft-solver/assets/10631777/f0a00b9c-949b-4d9a-9a2e-148517ea2eaa)

Would be nice to make it really efficient!

**Comment by vantezzen** [comment]
The biggest problem currently is that there is a large amount of recipes queued (<https://ics.vantezzen.io/status>) but the worker is relatively slow to fetch recipes.
This is because the worker is currently running a Chrome instance and sending requests directly from the page - if I tried directly requesting it in NodeJS with fetch the API will only return 403 forbidden.

So if you want to look at the code, feel free to search for a better way to fetch recipes without running a headless Chrome or if there are any other ways we can get recipes.
I thought about adding a feature where users could upload their discovered recipes from the game state but unfortunately Infinite Craft doesn't store that.

**Comment by NikitaMGrimm** [comment]
Infinite Craft might not save the recipes but its possible to extract them from the Firefox Cache.(located at about:cache?storage=disk) (Possibly other browsers, too)

I had a regex abomination to extract the recipes but it was error prone and I can't find it anymore.

**Comment by Lash-L** [comment]
@vantezzen Fwiw, my initial testing shows that the api will respond so long as you have a valid User agent 

This works for me in python:

```
f = requests.get("https://neal.fun/api/infinite-craft/pair?first=Cat&second=Dog",
                 headers={
    'Referer': 'https://neal.fun/infinite-craft/',
    'User-Agent': 'Solver'
})

```

I'm not really a big javascript guy otherwise I'd put forward a PR

**Comment by Tejs1** [comment]
> @vantezzen Fwiw, my initial testing shows that the api will respond so long as you have a valid User agent
> 
> This works for me in python:
> 
> ```
> f = requests.get("https://neal.fun/api/infinite-craft/pair?first=Cat&second=Dog",
>                  headers={
>     'Referer': 'https://neal.fun/infinite-craft/',
>     'User-Agent': 'Solver'
> })
> ```
> 
> I'm not really a big javascript guy otherwise I'd put forward a PR

Nope It doesn't work you might got response from disk cache
I tried same with puppeteer but not able to run it in headless mode @vantezzen How are you running it in headless.
Also @banneux-florent issue I am working on returning graph with least amount of nodes, edges

**Comment by Lash-L** [comment]
> > @vantezzen Fwiw, my initial testing shows that the api will respond so long as you have a valid User agent
> 
> > 
> 
> > This works for me in python:
> 
> > 
> 
> > ```
> 
> > f = requests.get("https://neal.fun/api/infinite-craft/pair?first=Cat&second=Dog",
> 
> >                  headers={
> 
> >     'Referer': 'https://neal.fun/infinite-craft/',
> 
> >     'User-Agent': 'Solver'
> 
> > })
> 
> > ```
> 
> > 
> 
> > I'm not really a big javascript guy otherwise I'd put forward a PR
> 
> 
> 
> Nope It doesn't work you might  have gotten response from disk cache
> 
> I tried same with puppeteer but not able to run it in headless mode @vantezzen How are you running it in headless.
> 
> Also @banneux-florent issue I am working on returning graph with least amount of nodes, edges 

I ran it in Python doing a brute force on word combos I had never done and in an empty script.just running the listed python code in a for loop.

**Comment by banneux-florent** [comment]
> > @vantezzen Fwiw, my initial testing shows that the api will respond so long as you have a valid User agent
> > This works for me in python:
> > ```
> > f = requests.get("https://neal.fun/api/infinite-craft/pair?first=Cat&second=Dog",
> >                  headers={
> >     'Referer': 'https://neal.fun/infinite-craft/',
> >     'User-Agent': 'Solver'
> > })
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I'm not really a big javascript guy otherwise I'd put forward a PR
> 
> Nope It doesn't work you might have gotten response from disk cache I tried same with puppeteer but not able to run it in headless mode @vantezzen How are you running it in headless. Also @banneux-florent issue I am working on returning graph with least amount of nodes, edges

How could we verify it's not from cache or so? Because I just ran the script and it's working for me too.

Today I tried making a local website calling the API but struggled to work around the CORS security so bad. This python thing working looks odd!

**Comment by Lash-L** [comment]
> How could we verify it's not from cache or so? Because I just ran the script and it's working for me too.
> 
> 
> 
> Today I tried making a local website calling the API but struggled to work around the CORS security so bad. This python thing working looks odd!

Just based off of the nature of python requests I can confirm it is not using a cache. Unless a server cache is being talked about out. In which case it is as none of my messages were new discoveries. But the api call id the same for new discoveries(words that no one has ever created before) so I would be extremely surprised if it didn't work. 

As for it not working for you in JavaScript, different libraries and languages can be very different internally in terms of the actually request. I'd check what headers are being sent, there's a chance one is being included that is getting blocked server side.

**Comment by banneux-florent** [comment]
> Just tested in Javscript, this works fine for me:
> 
> ```
> fetch('https://neal.fun/api/infinite-craft/pair?first=Cat&second=Dog', {
>   method: 'GET',
>   headers: {
>     'Referer': 'https://neal.fun/infinite-craft/',
>     'User-Agent': 'Solver'
>   }
> })
>   .then(response => response.json())
>   .then(data => {
>     // Handle the response data here
>     console.log(data);
>   });
> ```
(Missing code lines guessed and added)

This doesn't seems to work for me when trying from other websites or localhost. Where did you tried it?

**Comment by vantezzen** [comment]
Can confirm that it seems to work when adding a "Referer" header and having any valid User Agent now - maybe the security rules got updated there to be a bit more open. I tried it with Insomnia and NodeJS.

@banneux-florent This won't work from other websites due to CORS - you'll have to run the request from a priviledged environment.

**Comment by banneux-florent** [comment]
> Can confirm that it seems to work when adding a "Referer" header and having any valid User Agent now - maybe the security rules got updated there to be a bit more open. I tried it with Insomnia and NodeJS.
> 
> @banneux-florent This won't work from other websites due to CORS - you'll have to run the request from a priviledged environment.

Yes, I just finished making my VueJS app calling my PHP script that is itself calling the neal.fun API and it seems to work good!

**Comment by banneux-florent** [comment]
@vantezzen just sent you a message on Twitter/X. Sorry for notification, I'll delete this comment, but I don't know if it's got in spam or so. I wanted to ask you something!
